### ISSUE TITLE
write layer SBOMs to SBOM files instead of adding to layer and launch metadata

### ISSUE THREAD

**Issue by fg-j** [paketo-buildpacks/packit#253]
In order to implement [RFC 0038 – Support for CycloneDX and Syft SBoM in Paketo](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0038-cdx-syft-sbom.md), packit needs to update how it handles SBOM metadata. The high level change is that SBOM information ends up in its own files, instead of tagging along in files like `layer.toml`, `launch.toml` and `build.toml`.

- [`packit.Layer`](https://github.com/paketo-buildpacks/packit/blob/a1e3871627ca83315f74d46f12a5db5caccbd382/layer.go#L14) needs to have a BOM field added to hold on to layer-level BOM content
- If buildpack API version >= 0.7, `packit.Build()` should write the contents of a layer.BOM to a file(s) as JSON **if such a layer SBOM file doesn't already exist**. For lower API versions, packit should fail if layer.BOM has contents, since layer-level SBOMs aren't supported in earlier API versions
    -  Buildpacks will specify which BOM formats to generate via the `sbom` field in their `buildpack.toml` (see [this PR](https://github.com/paketo-buildpacks/packit/pull/248)). *For now, we need only support syft SBOM.*
    - For each specified format, for a layer named `output` created by the `cool-buildpack` this file needs to be placed at `/layers/cool-buildpack/output.sbom.<format>.json` 
   - the syft JSON needs to conform to the following schema
```
{
 "artifacts": [
  {
   "id": "8562221329663252200",
   "name": "bellsoft-java11",
   "version": "11.0.2",
   "licenses": ["GPL-2.0 WITH Classpath-exception-2.0"],
   "cpes": [
    "cpe:2.3:a:oracle:jre:11.0.2:*:*:*:*:*:*:*"
   ],
   "purl": "pkg:generic/bellsoft-java11@11.0.2?arch=amd64"
  }
 ],
 "descriptor": {
  "name": "syft",
  "version": "0.29.0"
 },
 "schema": {
  "version": "1.1.0",
  "url": "https://raw.githubusercontent.com/anchore/syft/main/schema/json/schema-1.1.0.json"
 }
}
```
-  `packit.Build()` should write the contents of `buildResult.Launch.BOM` to `/layers/cool-buildpack/launch.sbom.syft.json` if and only if buildpack API >= 0.7 **and a file at that location doesn't already exist**; for lower APIs, the launch BOM should still be added to `launch.toml` as it is currently doing
-  `packit.Build()` should write the contents of `buildResult.Build.BOM` to `/layers/cool-buildpack/build.sbom.syft.json` if and only if buildpack API >= 0.7  **and a file at that location doesn't already exist**; for lower APIs, the launch BOM should still be added to `launch.toml` as it is currently doing

### Open questions
- what is the correct value of `id` for a given artifact?

### Future work
- support generating cyclonedx SBOM (conforming to https://cyclonedx.org/docs/1.3/json)

**Comment by sophiewigmore** [comment]
check out the changes in https://github.com/paketo-buildpacks/packit/pull/241. It might be worthwhile to combine changes from this issue and that PR.

**Comment by fg-j** [comment]
Blocked on https://github.com/paketo-buildpacks/packit/pull/255 – this offers an alternate approach to supporting API v0.7 in packit.

**Comment by fg-j** [comment]
This is superceded by #255
